src/sentry/utils/appleconnect/__init__.py

```diff
@@ -0,0 +1,86 @@
+import datetime
+from collections import namedtuple
+from typing import Optional
+
+import requests
+
+from sentry.models.project import Project
+from sentry.utils import fernet_encrypt as encrypt
+from sentry.utils import json
+
+from . import appstore_connect, itunes_connect
+
+# We don't know the validity for sure, but we estimate 2 weeks
+TOKEN_VALIDITY = datetime.timedelta(weeks=2)
+
+AppConnectCredentialValidity = namedtuple(
+    "AppConnectCredentialValidity",
+    ["appstore_credentials_valid", "itunes_session_valid", "expiration_date"],
+)
+
+
+def credentials_key_name():
+    return "sentry:appleconnect_key"
+
+
+def symbol_sources_prop_name():
+    return "sentry:symbol_sources"
+
+
+def get_app_store_credentials(
+    project: Project, credentials_id: Optional[str]
+) -> Optional[json.JSONData]:
+    sources_config = project.get_option(symbol_sources_prop_name())
+
+    if credentials_id is None:
+        return None
+    try:
+        sources = json.loads(sources_config)
+        for source in sources:
+            if (
+                source.get("type") == "appStoreConnect"
+                and source.get("id") == credentials_id.lower()
+            ):
+                return source
+        return None
+    except BaseException as e:
+        raise ValueError("bad sources") from e
+
+
+def validate_credentials(
+    project: Project, credentials_id: Optional[str]
+) -> Optional[AppConnectCredentialValidity]:
+    credentials = get_app_store_credentials(project, credentials_id)
+    key = project.get_option(credentials_key_name())
+
+    if key is None or credentials is None:
+        return None
+
+    expiration_date = None
+    if credentials.get("refreshDate") is not None:
+        expiration_date = (
+            datetime.datetime.fromisoformat(credentials.get("refreshDate")) + TOKEN_VALIDITY
```

Comment:
This isn't supported on python 3.6.  You'll need something like `datetime.datetime.strptime(credentials.get("refershDate"), '%Y-%m-%d')`

Reply:
apparently you also have time with sub-second precision.  this removes the sub-second precision:
`datetime.datetime.strptime(credentials.get("refreshDate").split('.')[0], '%Y-%m-%dT%H:%M:%S')`

Reply:
or https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.isoparse apparently.  seems to already be in the dependencies

Reply:
yes we already have dateutil so in order to support all formats for isodate ( with or without time, millisecs... )
we should use:
```
    from dateutil import parser
    # ...
    my_date = parser.isoparse("2020-10-22")
```